packages/aws-cdk/lib/docker.ts

```diff
@@ -145,35 +142,22 @@ async function calculateImageFingerprint(imageId: string) {
   const manifestString = await shell(['docker', 'inspect', imageId], { quiet: true });
   const manifest = JSON.parse(manifestString)[0];
 
-  // Id can change
-  delete manifest.Id;
-
-  // Repository-based identifiers are out
-  delete manifest.RepoTags;
-  delete manifest.RepoDigests;
-
-  // Metadata that has no bearing on the image contents
-  delete manifest.Created;
-
-  // We're interested in the image itself, not any running instaces of it
-  delete manifest.Container;
-  delete manifest.ContainerConfig;
-
-  // We're not interested in the Docker version used to create this image
-  delete manifest.DockerVersion;
-
-  // On some Docker versions Metadata contains a LastTagTime which updates
-  // on every push, causing us to miss all cache hits.
-  delete manifest.Metadata;
-
-  // GraphDriver is about running the image, not about the image itself.
-  delete manifest.GraphDriver;
+  // It's quite likely the only "actually" relevant field is "RootFS" (hashes of the layers), but
```

Comment:
See #2072 - maybe we are doing something wrong here

Reply:
Regarding #2072 (PR not opened by me but where I commented), I still think that using the repo digest as suggested in https://github.com/awslabs/aws-cdk/pull/2072#issuecomment-475272427 should be considered and may be better than the current fingerprint approach.

Reply:
@jogold I had considered this but I was rather unsure of the stability of repository digests. However I stand convinced that this is a safer approach (would rather build once too many than not re-build when I should have). The upcoming revision will use the repository tag.